File: app/src/main/java/com/nedaluof/qurany/data/api/ApiService.java
package com.nedaluof.qurany.data.api;

import com.nedaluof.qurany.data.model.Reciters;

import io.reactivex.Observable;
import retrofit2.http.GET;
import retrofit2.http.Path;

/**
 * Created by nedaluof on 6/13/2020.
 */
public interface ApiService {

    /**
     * @return Observable Reciters obj
     */
    @GET("{language}.php")
    Observable<Reciters> getReciters(
            @Path("language") String language
    );

}

File: app/src/main/java/com/nedaluof/qurany/data/source/remotesource/api/ApiService.kt
package com.nedaluof.qurany.data.source.remotesource.api

import com.nedaluof.qurany.data.model.Reciters
import retrofit2.Response
import retrofit2.http.GET
import retrofit2.http.Path

/**
 * Created by nedaluof on 12/11/2020.
 */
interface ApiService {

  @GET("{language}.php")
  suspend fun getReciters(
    @Path("language") language: String
  ): Response<Reciters>

  companion object {
    const val BASE_URL = "http://mp3quran.net/api/"
  }
}
